mu-plugins/pwcc-helpers/inc/tachyon-mods/namespace.php

```diff
@@ -0,0 +1,463 @@
+			if ( strpos( $gravity, 'east' ) !== false ) {
+				$gravity = 'east';
+			} elseif ( strpos( $gravity, 'west' ) !== false ) {
+				$gravity = 'west';
+			}
+			break;
+	}
+
+	$tachyon_args['gravity'] = $gravity;
+	return $tachyon_args;
+}
+
+/**
+ * Filters 'img' elements in post content to add 'srcset' and 'sizes' attributes.
+ *
+ * @param string $content The raw post content to be filtered.
+ *
+ * @return string Converted content with 'srcset' and 'sizes' attributes added to images.
+ */
+function make_content_images_responsive( $content ) {
+	$images = \Tachyon::parse_images_from_html( $content );
+
+	if ( empty( $images ) ) {
+		// No images, leave early.
+		return $content;
+	}
+
+	// This bit is from Core.
+	$selected_images = [];
+	$attachment_ids = [];
+	foreach ( $images['img_tag'] as $image ) {
+		if ( false === strpos( $image, ' srcset=' ) && preg_match( '/wp-image-([0-9]+)/i', $image, $class_id ) && absint( $class_id[1] ) ) {
+			$attachment_id = $class_id[1];
+			/*
+			 * If exactly the same image tag is used more than once, overwrite it.
+			 * All identical tags will be replaced later with 'str_replace()'.
+			 */
+			$selected_images[ $image ] = $attachment_id;
+			// Overwrite the ID when the same image is included more than once.
+			$attachment_ids[ $attachment_id ] = true;
+		}
+	}
+
+	if ( empty( $attachment_ids ) ) {
+		// No WP attachments, nothing further to do.
+		return $content;
+	}
+
+	/*
+	 * Warm the object cache with post and meta information for all found
+	 * images to avoid making individual database calls.
+	 */
+	_prime_post_caches( array_keys( $attachment_ids ), false, true );
+
+	foreach ( $images['img_url'] as $key => $img_url ) {
+		if ( strpos( $img_url, TACHYON_URL ) !== 0 ) {
+			// It's not a Tachyon URL.
+			continue;
+		}
+
+		$image_data = [
+			'full_tag' => $images[0][ $key ],
+			'link_url' => $images['link_url'][ $key ],
+			'img_tag' => $images['img_tag'][ $key ],
+			'img_url' => $images['img_url'][ $key ],
+		];
+		$image = $image_data['img_tag'];
+
+		$attachment_id = $selected_images[ $images['img_tag'][ $key ] ];
+		$image_meta = wp_get_attachment_metadata( $attachment_id );
+		$content = str_replace( $image, add_srcset_and_sizes( $image_data, $image_meta, $attachment_id ), $content );
+	}
+
+	return $content;
+}
+
+/**
+ * Adds 'srcset' and 'sizes' attributes to an existing 'img' element.
+ *
+ * @TODO Deal with crop.
+ *
+ * @param array $image_data    The full data extracted via `make_content_images_responsive`.
+ * @param array $image_meta    The image meta data as returned by 'wp_get_attachment_metadata()'.
+ * @param int   $attachment_id Image attachment ID.
+ *
+ * @return string Converted 'img' element with 'srcset' and 'sizes' attributes added.
+ */
+function add_srcset_and_sizes( $image_data, $image_meta, $attachment_id ) {
+	$image = $image_data['img_tag'];
+	$image_src = $image_data['img_url'];
+
+	// Bail early if an image has been inserted and later edited.
```

Comment:
TODO: Switch to old sizing data stored in post meta as `_wp_attachment_backup_sizes`